### ISSUE TITLE
Parallel test execution with TestNG is not rendered in console

### ISSUE THREAD

**Issue by bmuschko** [gradle/gradle#2786]
### Expected Behavior

The console renders parallel test execution in the console the same way as for JUnit.

__build.gradle__

```
apply plugin: 'java'

repositories {
    jcenter()
}

dependencies {
    testImplementation 'junit:junit:4.12'
}

test {
    maxParallelForks = 4
}
```

__Console output:__

```
<===========--> 87% EXECUTING [3s]
> :test > 44 tests completed
> :test > Executing test MyTest109
> :test > Executing test MyTest108
> :test > Executing test MyTest106
> :test > Executing test MyTest107
```

### Current Behavior

Executing tests with TestNG only renders the test execution summary in the console but no individual test classes.

__build.gradle__

```
apply plugin: 'java'

repositories {
    jcenter()
}

dependencies {
    testImplementation 'org.testng:testng:6.11'
}

test {
    useTestNG()
    maxParallelForks = 4
}
```

__Console output:__

```
<===========--> 87% EXECUTING [4s]
> :test > 85 tests completed
```

### Context

The console needs to behave the same way independent of the used JVM test framework.

### Steps to Reproduce (for bugs)

see examples above

**Comment by bmuschko** [comment]
The proper fix would emit a `DefaultTestClassDescriptor` event from test execution with TestNG. Right now this is only happening for JUnit. While the proposed PR works functionally, it causes a performance regression.

**Comment by bmuschko** [comment]
I implemented a spike in the branch [`bm/console/parallel-testng`](https://github.com/gradle/gradle/compare/bm/console/parallel-testng). It turns out to be a time sink and I am not going to spend more time until it is prioritized. The problem with the implementation is that TestNG does not provide a listener for events like "start test class" and "finish test class". You kind of have to derive the information from other events which is error prone and tedious. The current implement works functionally but I already ran into a couple of corner cases that would have to be covered properly e.g. included/excluded test methods etc.
